Automattic/jetpack
# Show custom Boost Tooltip content based on letter grade and card state

This PR displays different Boost score Tooltip **copy** depending on the Boost score letter grade and other factors (such as: is Boost plugin active or not?, and does the site have a paid boost plan or not?).

Related to: _PT: My Jetpack: Add Boost speed score into boost product card_: pbNhbs-9Nt-p2
Fixes: https://github.com/Automattic/jetpack-roadmap/issues/1269

The **Tooltip copy doc** (source of truth): 33a29-pb

**Important note:** This PR does not display different copy based on if there is/was an _increase_  (_or decrease_) in Boost scores (as outlined in the Tooltip copy doc, above). All copy shown in this PR, is the copy for when there is no increase is the Boost speed scores. I will add the copy & logic for when there is an _increase_ in Boost speed scores in a separate follow-up PR.




 
**Screenshots:**

**Boost is not active and/or site is not connected:**

![Markup 2024-02-22 at 08 37 52](https://github.com/Automattic/jetpack/assets/11078128/b035218e-ee2e-47a0-bf31-63630a1e575c)

![Screenshot on 2024-02-22 at 08-38-58](https://github.com/Automattic/jetpack/assets/11078128/9cc05c67-de49-4a8f-b620-51f7fb86ba56)

![Screenshot on 2024-02-22 at 08-41-59](https://github.com/Automattic/jetpack/assets/11078128/c8bbb639-a90b-4e14-94f2-1dcf1fee69f7)


**Boost is active - has Free plan**

![Screenshot on 2024-02-22 at 09-34-53](https://github.com/Automattic/jetpack/assets/11078128/d3bd139f-d2ad-4862-ad15-8bc41b3c8350)


![Screenshot on 2024-02-22 at 09-35-54](https://github.com/Automattic/jetpack/assets/11078128/8cf57c90-1b81-44f8-abe2-70e896347c1d)

![Screenshot on 2024-02-22 at 09-36-52](https://github.com/Automattic/jetpack/assets/11078128/31184cbf-5292-4ff1-8e28-9776a6fd98ba)

**Boost is Active - has Paid plan**

![Screenshot on 2024-02-22 at 09-30-43](https://github.com/Automattic/jetpack/assets/11078128/61414e69-c735-493b-904c-6e4bcf10e6b1)

![Screenshot on 2024-02-22 at 09-31-42](https://github.com/Automattic/jetpack/assets/11078128/001eca40-2c9c-4a5e-9794-dd0776daf98d)

![Screenshot on 2024-02-22 at 09-33-21](https://github.com/Automattic/jetpack/assets/11078128/af8aec24-dfba-4ab1-bf1a-71a2a69ce5c9)


### Other information:


## Jetpack product discussion



Github project task: [Add value to the active Boost product card on My Jetpack](https://github.com/orgs/Automattic/projects/702/views/6?pane=issue&itemId=50279848)

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch using the Jetpack Beta plugin, Or using your local dev environment.
* Without Jetpack connected yet, hover over the Boost score letter grade. Verify the tooltip copy matches the correct copy listed in the Tooltip copy doc (uninstalled state).
* Connect Jetpack & activate Boost free
* Hover over the Boost score letter grade and verify the tooltip copy matches the correct copy listed in the Tooltip copy doc (Active State - when there is no increase - Free users).
* Next Purchase a Boost **Paid** plan:
* Hover over the Boost score letter grade and verify the tooltip copy matches the correct copy listed in the Tooltip copy doc (Active State - when there is no increase - Paid users).
* Inspect the code. Verify the logic looks correct, displaying the correct letter grade copy, based on whether the plugin is 'Active' or not and whether the site has a paid Boost subscription or not.
    - Note: In this PR we do not include copy for when there is an increase in Boost speed scores, as outlined in the Tooltip copy doc. I will add this in a separate follow-up PR.
